### PR TITLE
app/vmalert: show on UI groups error after reload config

### PR DESCRIPTION
**Pull request by dmitryk-dk**
Show the last error message received during config reload in vmalert's UI header. Please check the image below
Screenshot 2023-06-30 at 09 50 09

Related issue: https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4076

Signed-off-by: dmitryk-dk [d.kozlov@victoriametrics.com](mailto:d.kozlov@victoriametrics.com)

### CODE REVIEW

File: app/vmalert/manager.go [hunk 1]
```diff
@@ -26,8 +26,9 @@ type manager struct {
 	wg     sync.WaitGroup
 	labels map[string]string
 
-	groupsMu sync.RWMutex
-	groups   map[uint64]*Group
+	groupsMu        sync.RWMutex
+	groups          map[uint64]*Group
+	groupsReloadErr error
```
Comment (hagen1778) [inline app/vmalert/manager.go:31]: Could this field be a global variabe? Is there a reason why it should belong to manager object?

**Review by hagen1778 — CHANGES_REQUESTED** [review]

File: app/vmalert/manager.go (hunk 1 above)
Comment (dmitryk-dk) [reply app/vmalert/manager.go]: used global variabel and renamed it to userError

File: app/vmalert/tpl/header.qtpl [hunk 2]
```diff
@@ -6,7 +6,7 @@
     "github.com/VictoriaMetrics/VictoriaMetrics/app/vmalert/utils"
 ) %}
 
-{% func Header(r *http.Request, navItems []NavItem, title string) %}
+{% func Header(r *http.Request, navItems []NavItem, title string, groupsError error) %}
```
[the remark this answers is not in the document]
Comment (dmitryk-dk) [reply app/vmalert/tpl/header.qtpl]: changed name

File: app/vmalert/web.go [hunk 3]
```diff
@@ -328,6 +328,10 @@ func (rh *requestHandler) alertByPath(path string) (*APIAlert, error) {
 	return resp, nil
 }
 
+func (rh *requestHandler) groupsReloadError() error {
```
[the remark this answers is not in the document]
Comment (dmitryk-dk) [reply app/vmalert/web.go]: removed because used gloabl var

File: app/vmalert/web.qtpl [hunk 4]
```diff
@@ -12,7 +12,7 @@
 
 
 {% func Welcome(r *http.Request) %}
-    {%= tpl.Header(r, navItems, "vmalert") %}
+    {%= tpl.Header(r, navItems, "vmalert", nil) %}
```
[the remark this answers is not in the document]
Comment (dmitryk-dk) [reply app/vmalert/web.qtpl]: added support for all pages

File: docs/CHANGELOG.md [hunk 5]
```diff
@@ -35,6 +35,7 @@ The following tip changes can be tested by building VictoriaMetrics components f
...
 * FEATURE: [vmalert](https://docs.victoriametrics.com/vmalert.html): expose `vmalert_remotewrite_send_duration_seconds_total` counter, which can be used for determining high saturation of every connection to remote storage with an alerting query `sum(rate(vmalert_remotewrite_send_duration_seconds_total[5m])) by(job, instance) > 0.9 * max(vmalert_remotewrite_concurrency) by(job, instance)`. This query triggers when a connection is saturated by more than 90%. This usually means that `-remoteWrite.concurrency` command-line flag must be increased in order to increase the number of concurrent writings into remote endpoint. See [this feature request](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4516).
+* FEATUTE: [vmalert](https://docs.victoriametrics.com/vmalert.html): show the last error message received during config reload in vmalert's UI header.
```
[the remark this answers is not in the document]
Comment (dmitryk-dk) [reply docs/CHANGELOG.md]: fixed

File: app/vmalert/tpl/header.qtpl [hunk 6]
```diff
@@ -103,5 +103,16 @@ type NavItem struct {
             {% endfor %}
         </ul>
   </div>
+      {% if groupsError != nil %}
```
[the remark this answers is not in the document]
Comment (dmitryk-dk) [reply app/vmalert/tpl/header.qtpl]: created user error snippet, split it into icon and content.

**Review by hagen1778 — APPROVED** [review]
Thanks!

**Pull request merged**